thrift/thrift-gen/main.go

```diff
@@ -149,3 +200,67 @@ func cleanGeneratedCode(generated []byte) []byte {
 func contextType() string {
 	return "thrift.Context"
 }
+
+func getTemplateFromFile(filename string, AST *parser.Thrift) (*template.Template, error) {
+	// export useful functions to the template renderer
+	state := NewState(AST)
+	funcs := map[string]interface{}{
+		"contextType": contextType,
+		"lowercase": func(in string) string {
+			return strings.ToLower(in)
+		},
+		"goPublicName": goPublicName,
+		"goType": func(t *parser.Type) string {
+			return state.goType(t)
+		},
+	}
+
+	// read template file
+	templateBytes, err := ioutil.ReadFile(filename)
```

Comment:
I thought we decided to do paths relative to the `GOPATH` rather than absolute paths?